### PR TITLE
Update/refactor test/integration/user-management-api/account_plan_features_test

### PR DESCRIPTION
**Pull request by josemigallas**
♻️ refactor a lot
✅ fixes deprecation params as positional arg
🚨 fixes lint (rubocop)
🎨 improves code format

### CODE REVIEW

File: test/integration/user-management-api/account_plan_features_test.rb [hunk 1]
```diff
@@ -25,52 +26,37 @@ def setup
...
-              :feature_id => feat.id,
-              :provider_key => @provider.api_key, :format => :xml)
-
+    post admin_api_account_plan_features_path(@account_plan.id), params: { feature_id: feat.id, provider_key: @provider.api_key, format: :xml }
     assert_response :success
 
     xml = Nokogiri::XML::Document.parse(@response.body)
-
-    assert xml.xpath('.//feature/id').children.first.text == feat.id.to_s
+    assert_equal xml.xpath('.//feature/id').children.first.text, feat.id.to_s
```
Comment (akostadinov) [inline test/integration/user-management-api/account_plan_features_test.rb:55]: This seems to be safe here. But just FYI `#assert_equal` is not always safe. See #2627

File: test/integration/user-management-api/account_plan_features_test.rb [hunk 2]
```diff
@@ -25,52 +26,37 @@ def setup
...
   pending_test 'enable existing feature'
 
   test 'disable feature' do
-    feat = FactoryBot.create :feature, :featurable => @provider
-
-    post(admin_api_account_plan_features_path(@account_plan.id),
-              :feature_id => feat.id,
-              :provider_key => @provider.api_key, :format => :xml)
-
+    post admin_api_account_plan_features_path(@account_plan.id), params: { feature_id: feat.id, provider_key: @provider.api_key, format: :xml }
```
Comment (akostadinov) [inline test/integration/user-management-api/account_plan_features_test.rb:51]: Just curious, how are we disabling anything here? In docs I see this as creating.
image

**Review by akostadinov — APPROVED** [review]
One question. I assume everything is alright so approving anyway.

File: test/integration/user-management-api/account_plan_features_test.rb [hunk 3]
```diff
@@ -25,52 +26,37 @@ def setup
...
     xml = Nokogiri::XML::Document.parse(@response.body)
-
-    assert xml.xpath('.//feature/id').children.first.text == feat.id.to_s
+    assert_equal xml.xpath('.//feature/id').children.first.text, feat.id.to_s
   end
 
-  pending_test 'disable non-existing feature'
-
+  test 'disable non-existing feature' do
+    post admin_api_account_plan_features_path(@account_plan.id), params: { feature_id: 'nope', provider_key: @provider.api_key, format: :xml }
```
Comment (hallelujah) [inline test/integration/user-management-api/account_plan_features_test.rb:59]: Shouldn't it be `nil` or empty string `''` instead?

File: test/integration/user-management-api/account_plan_features_test.rb (hunk 2 above)
Comment (hallelujah) [reply test/integration/user-management-api/account_plan_features_test.rb:51]: Well create and enable are basically synonym in this context. I think doc is a bit confusing like you, but it is to create a link between a feature and a plan

File: test/integration/user-management-api/account_plan_features_test.rb (hunk 3 above)
Comment (josemigallas) [reply test/integration/user-management-api/account_plan_features_test.rb:59]: Do you mean `feature_id`? I think those are as good options as an inexistent ID, no?

Comment (hallelujah) [reply test/integration/user-management-api/account_plan_features_test.rb:59]: Ah! well why not, not sure :)

File: test/integration/user-management-api/account_plan_features_test.rb (hunk 2 above)
Comment (akostadinov) [reply test/integration/user-management-api/account_plan_features_test.rb:51]: I was referring to test name `disable feature`. But I don't see us disable anything. That's why I asked. Maybe test name better be changed?

> doc is a bit confusing like you

Apologizes, fwiw my gf thinks the same :angel:

File: test/integration/user-management-api/account_plan_features_test.rb (hunk 1 above)
Comment (josemigallas) [reply test/integration/user-management-api/account_plan_features_test.rb:55]: It looks to me that they're exactly the same:
https://github.com/seattlerb/minitest/blob/f4f57afaeb3a11bd0b86ab0757704cb78db96cf4/lib/minitest/assertions.rb#L218

File: test/integration/user-management-api/account_plan_features_test.rb (hunk 2 above)
Comment (josemigallas) [reply test/integration/user-management-api/account_plan_features_test.rb:51]: It's very confusing yes, I even debugged a little and didn't found anything "disabled"... ultimately decided not to change it and just fix the deprecation. We can take a look at it again in the future.

**Review by jsmadis — APPROVED** [review]

**Review by hallelujah — APPROVED** [review]

**Pull request merged**